python/looker_sdk/rtl/api_settings.py

```diff
@@ -32,13 +32,21 @@
 from looker_sdk.rtl import transport
 
 if sys.version_info >= (3, 8):
-    from typing import Protocol
+    from typing import Protocol, TypedDict
 else:
-    from typing_extensions import Protocol
+    from typing_extensions import Protocol, TypedDict
 
+from typing_extensions import Required
```

Comment:
I tried with 3.11.0a4 but `Required` wasn't in there.

So I've

1. kept the import coming from `typing_extensions`
2. updated setup.py to require typing-extensions >= 4.1.1 regardless of the python version

I think that is the best way? Note we still can't use 3.11 anyway because of #944 (I pulled Required from typing_extensions running 3.11 and got the error referenced in that issue).